### ISSUE TITLE
Use Inspec::UI consistently

### ISSUE THREAD

**Issue by clintoncwolfe** [inspec/inspec#3715]
### 🎛 Description

#3618 introduced a facility for standardizing UI interactions, but did not apply those changes to the InSpec codebase.

So... let's use Inspec::UI throughout the codebase.

Consider carefully any changes that would be breaking, and make those on a separate PR.

### 🌍 InSpec and Platform Version

3.2.x

### 🤔 Replication Case

List key points here?

### Related issues

#1700, #3648, #3580, #3556, #3513, #2384, #3818

**Comment by wyardley** [comment]
@clintoncwolfe thanks for this.
Question though, it doesn't seem like `lib/inspec/ui.rb` currently has any kind of support for color terminals that don't support 256 colors, so that even if we switched to this, #1700 would still be an issue, right?

**Comment by miah** [comment]
I've fixed most of these in #4000, but there are 3 remaining:

```
  2) Failure:
cli command plugins#test_0001_is able to respond to a plugin-based cli subcommand [/home/miah/projects/github/inspec_cli/test/functional/plugins_test.rb:63]:
--- expected
+++ actual
@@ -1 +1,2 @@
-""
+"[2019-04-30T11:15:39-07:00] WARN: DEPRECATION:  Please call `cli.ui` directly
+"


  3) Failure:
inputs::run profile with yaml inputs#test_0004_errors with required input not defined [/home/miah/projects/github/inspec_cli/test/functional/inputs_test.rb:103]:
--- expected
+++ actual
@@ -1 +1,2 @@
-""
+"[2019-04-30T11:19:49-07:00] WARN: DEPRECATION:  Please call `cli.ui` directly
+"


  4) Failure:
inputs::run profile with yaml inputs#test_0003_errors with invalid input types [/home/miah/projects/github/inspec_cli/test/functional/inputs_test.rb:93]:
--- expected
+++ actual
@@ -1 +1,2 @@
-""
+"[2019-04-30T11:19:49-07:00] WARN: DEPRECATION:  Please call `cli.ui` directly
+"

```

**Comment by wyardley** [comment]
I don't think I can reopen it myself; @clintoncwolfe is this still something that might happen someday/
